lighthouse-core/lib/traces/metrics-evts.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const log = require('../../../lighthouse-core/lib/log.js');
+
+class Metrics {
+
+  constructor(traceEvents, auditResults) {
+    this._traceEvents = traceEvents;
+    this._auditResults = auditResults;
+    this._fakeEvents = [];
+  }
+
+  /**
+   * Returns simplified representation of all metrics
+   * @return {!Array<{getTs: Function, id: string, name: string}>} metrics to consider
+   */
+  static get metricsDefinitions() {
+    return [
+      {
+        name: 'Navigation Start',
+        id: 'navstart',
+        getTs: auditResults => {
+          const fmpExt = auditResults['first-meaningful-paint'].extendedInfo;
+          return fmpExt.value.timestamps.navStart;
+        }
+      },
+      {
+        name: 'First Contentful Paint',
+        id: 'ttfcp',
+        getTs: auditResults => {
+          const fmpExt = auditResults['first-meaningful-paint'].extendedInfo;
```

Comment:
Essentially we have 3 actual audits in lighthouse that collect these metrics. And yeah we stash several key metrics within the extInfo of each of them. 
So, yeah, AFAIK all this is correct and intended... any ideas on how it could look less odd?